mslib/msui/mscolab.py

```diff
@@ -1001,23 +995,24 @@ def handle_update_permission(self, p_id, u_id, access_level):
 
             self.access_level = access_level
             # Close mscolab windows based on new access_level and update their buttons
-            if self.access_level == "collaborator" or self.access_level == "viewer":
-                self.adminWindowBtn.setEnabled(False)
-                self.versionHistoryBtn.setEnabled(False)
-                if self.admin_window is not None:
-                    self.admin_window.close()
-                if self.version_window is not None:
-                    self.version_window.close()
-            else:
-                self.adminWindowBtn.setEnabled(True)
-                self.versionHistoryBtn.setEnabled(True)
-
-            if self.access_level == "viewer":
-                self.chatWindowBtn.setEnabled(False)
-                if self.chat_window is not None:
-                    self.chat_window.close()
-            else:
-                self.chatWindowBtn.setEnabled(True)
+            allow_version_access = self.access_level in ["collaborator", "creator", "admin"]
+            self.versionHistoryBtn.setEnabled(allow_version_access)
+            self.chatWindowBtn.setEnabled(allow_version_access)
+            self.workLocallyCheckBox.setEnabled(allow_version_access)
+            self.importBtn.setEnabled(allow_version_access)
+            if not allow_version_access and self.version_window is not None:
+                self.version_window.close()
+            if not allow_version_access and self.chat_window is not None:
+                self.chat_window.close()
+            if not allow_version_access and self.handle_work_locally_toggle is not None:
+                self.handle_work_locally_toggle.close()
+            if not allow_version_access and self.handle_import is not None:
+                self.handle_import.close()
+
+            allow_version_access = self.access_level in ["creator", "admin"]
+            self.adminWindowBtn.setEnabled(allow_version_access)
+            if not allow_version_access and self.admin_window is not None:
+                self.admin_window.close()
```

Comment:
Much better!
It seems to me that in the second piece of code an update of "self.deleteProjectBtn" is missing?
If it is not missing, it is fine this way.
If the code needs to be touched again due to this, I would strongly suggest merging lines 731-741 and lines 998-1015 to a single function. The window closing may be superfluous in the first section, but the maintainability gain due to a single piece of code is huge.

Reply:
Great suggestion, I have merged both blocks of code to new function `def handle_mscolab_buttons(self):`